pkg/router/template/router.go

```diff
@@ -1318,9 +1318,13 @@ func (r *templateRouter) calculateServiceWeights(serviceUnits map[ServiceUnitKey
 	// If there is only 1 service unit, then always set the weight 1 for all the endpoints.
 	// Scaling the weight to 256 is redundant and causes haproxy to allocate more memory on startup.
 	if len(serviceUnits) == 1 {
-		for key := range serviceUnits {
+		for key, units := range serviceUnits {
```

Comment:
So I understand why you used `units` here, because of the other existing parts of the code use `units`, but to me, it's confusing (not your fault, the code is already a bit confusing IMO). What's a unit? Looks like `units` is actually a weight here for the service.

I think we should avoid propagating confusing variable names:

```suggestion
		for key, weight := range serviceUnits {
```